hive/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```diff
@@ -249,13 +257,32 @@ private long acquireLock() throws UnknownHostException, TException, InterruptedE
     LockResponse lockResponse = metaClients.run(client -> client.lock(lockRequest));
     LockState state = lockResponse.getState();
     long lockId = lockResponse.getLockid();
-    //TODO add timeout
+
+    final long start = System.currentTimeMillis();
+    long duration = 0;
+    boolean timeout = false;
     while (state.equals(LockState.WAITING)) {
       lockResponse = metaClients.run(client -> client.checkLock(lockId));
       state = lockResponse.getState();
+
+      // check timeout
+      duration = System.currentTimeMillis() - start;
+      if (duration > lockAcquireTimeout) {
+        timeout = true;
+        break;
```

Comment:
Instead of break, can we use `timeout` in the `while` condition?

```java
while (!timeout && state.equals(LockState.WAITING)) { ... }
```

As long as we are setting `timeout`, I think it is nice to use it to avoid control flow breaks. We would also need to move `Thread.sleep(50)` into an `else`.